experimental/single_number_formatter/src/ule.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerovec::ule::{AsULE, ZeroVecError, ULE};
+
+use crate::provider::{CurrencyPatterns, PatternSelection};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[repr(transparent)]
+pub struct CurrencyPatternsULE([u8; 3]);
+
+impl CurrencyPatternsULE {}
+
+// Safety (based on the safety checklist on the ULE trait):
+//  1. CurrencyPatternsULE does not include any uninitialized or padding bytes.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  2. CurrencyPatternsULE is aligned to 1 byte.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid.
+//  4. The impl of validate_byte_slice() returns an error if there are extra bytes.
+//  5. The other ULE methods use the default impl.
+//  6. CurrencyPatternsULE byte equality is semantic equality.
+unsafe impl ULE for CurrencyPatternsULE {
+    fn validate_byte_slice(bytes: &[u8]) -> Result<(), ZeroVecError> {
+        if bytes.len() % 3 != 0 {
+            return Err(ZeroVecError::length::<Self>(bytes.len()));
+        }
+
+        // #[allow(clippy::indexing_slicing)] // chunks
+        // if !bytes
+        //     .chunks(3)
+        //     .all(|c| Self::bytes_in_range((&c[0], &c[1], &c[2])))
+        // {
+        //     return Err(ZeroVecError::parse::<Self>());
+        // }
```

Comment:
observation: you're treating the bit values `01`, `11` as `::StandardAlphaNextToNumber`, and `00`, `10` as `::Standard`, so not validation is required here.